composer/datasets/streaming/dataset.py

```diff
@@ -209,14 +216,14 @@ def _insert_shard_samples(self, shard: int, part_min_id: int, part_max_id: int)
             # Extend and optionally reshuffle the remaining samples of any
             # epochs we have in progress.
             if self.shuffle:
-                if not self._is_downloaded:
+                if self._download_status == DownloadStatus.IN_PROGRESS:
                     self._downloaded_ids.extend(new_ids)
                     np.random.shuffle(self._downloaded_ids)
                 for todo_ids in self._epoch_to_todo_ids.values():
```

Comment:
Do we need to handle DownloadStatus == FAILED and DownloadStatus.NOT_STARTED?

Reply:
insert_shard_samples() is only called from download(), which does not call it in those states